files/ko/learn/tools_and_testing/client-side_javascript_frameworks/react_todo_list_beginning/index.md

```diff
@@ -0,0 +1,635 @@
+
+- 새로운 태스크를 쓰기 위한 [`<input type="text">`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/text)와 폼(form)을 제출하기 위한 버튼을 가진 [`<form>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form) 요소가 있습니다.
+- 태스크를 필터링하기 위한 여러 개의 버튼이 있습니다.
+- 태스크가 몇 개 남아있는지 알려주는 헤딩(heading)이 있습니다.
+- 순서가 매겨지지 않은 세 개의 태스크가 있습니다. 각각의 태스크는 리스트 아이템([`<li>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/li))이며, 편집하거나 삭제하기 위한 버튼과 완료를 표시할 체크 박스를 가지고 있습니다.
+
+폼은 태스크를 *만들* 수 있게 해줍니다; 버튼들은 태스크들을 *필터링할* 수 있게 해줍니다; 헤딩과 리스트는 태스크들을 *읽을* 수 있게 해줍니다. 태스크를 *편집하는* UI는 현재는 없습니다. 이것도 괜찮습니다 – 나중에 만들 거니까요.
+
+
+
+### 접근성 기능
+
+익숙하지 않은 속성을 발견했을 겁니다. 예를 들어:
+
+```js
+<button type="button" className="btn toggle-btn" aria-pressed="true">
+  <span className="visually-hidden">Show </span>
+  <span>all</span>
+  <span className="visually-hidden"> tasks</span>
+</button>
+```
+
+`aria-pressed`은 (스크린 리더와 같은) assistive technology에게 하나의 버튼이 두 개의 상태 `pressed`와 `unpressed`로 있을 수 있다는 것을 알려줍니다. `on`과 `off`라고 생각해보세요. 값을 `true`로 설정하는 것은 버튼을 기본적으로 눌린 상태라는 것을 의미합니다.
+
+클래스 `visually-hidden`는 아직 CSS를 연결하지 않았으므로 아직 아무런 이펙트도 없습니다. 제대로 스타일을 설정하면 이 클래스를 가진 모든 요소가 시력이 정상인 사용자에게는 숨겨지며 스크린 리더 사용자에게만 보입니다 — 이 단어들은 시력이 정상인 사용자들에게 필요하지 않기 때문입니다; 단어들은 추가적인 시각적 맥락(visual context)을 가지고 있지 않은 스크린 리더 사용자들에게 버튼이 무엇을 하는지 더 많은 정보를 제공합니다.
+
+아래에서 [`<ul>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul) 요소를 찾을 수 있을 겁니다.
+
+```js
+<ul
+  role="list"
+  className="todo-list stack-large stack-exception"
+  aria-labelledby="list-heading"
+>
+```
+
+`role` 속성은 assistive technology가 태그가 나타내고 있는 요소의 종류가 무엇인지 설명하는 데 도움이 됩니다. `<ul>`은 기본적으로 리스트처럼 취급되지만, 우리가 추가할 스타일은 리스트의 기능성을 해칩니다. `role="list"`는 `<ul>` 요소가  "리스트"의 의미를 회복하도록 합니다. 이것이 왜 필요한지 알고 싶다면, [Scott O'Hara’s article, “Fixing Lists”](https://www.scottohara.me/blog/2019/01/12/lists-and-safari.html)을 참고하세요.
+
+`aria-labelledby` 속성은 assistive technologies에게 리스트 헤딩을 헤딩 아래의 리스트의 목적을 나타내는 라벨로 사용하고 있음을 알립니다. 이러한 연관을 만드는 것은 리스트가 더욱 정보가 많은 맥락을 가지게 하여, 스크린 리더 사용자가 리스트의 목적을 더 잘 이해할 수 있도록 합니다.
+
+마지막으로, 리스트 항목 안의 라벨과 인풋들은 JSX에 고유한 속성을 가지고 있습니다.
+
+```js
+<input id="todo-0" type="checkbox" defaultChecked={true} />
+<label className="todo-label" htmlFor="todo-0">
+  Eat
+</label>
+```
+
+`<input/ >`의 `defaultChecked` 속성은 React가 초기에 이 체크박스를 체크하도록 합니다. HTML에서처럼 `checked`를 사용한다면, React는 체크박스에 대한 이벤트를 다루는 것과 관련한 경고를 브라우저 콘솔에 출력할 것입니다. 지금은 크게 걱정하지 마세요 — 나중에 이벤트를 사용할 때 이야기할테니까요.
+
+`htmlFor` 속성은 HTML에서는 `for` 속성과 대응합니다. JSX에서는 `for`을 속성으로 사용하지 않는데, `for`이 예약어(reserved word)이기 때문입니다. 그래서 React는 `htmlFor`을 대신 사용합니다.
+
+참고:
+
+- JSX 속성에 불리언 값들(`true`와 `false`)를 사용하기 위해서는 이 값들을 중괄호로 감싸야 합니다. `defaultChecked="true"`라고 적는다면, `defaultChecked`의 값은 문자열 리터럴인 `"true"`가 될 겁니다. 꼭 기억하세요 — JSX는 JavaScript이지, HTML이 아닙니다!
+- 이전의 코드 스니펫에서 사용되었던 `aria-pressed` 속성은 `"true"`라는 값을 가지는데, 이것은 `aria-pressed`가 `checked`가 있는 방식에서 진짜 불리언 속성이 아니기 때문입니다(because `checked` is not a true boolean attribute in the way `checked` is).
```

Comment:
`because aria-pressed is not a true boolean attribute in the way checked is.`인데 `because checked is not a true boolean attribute in the way checked is.`로 본문이 잘못 옮겨져 있습니다.

또한 충분히 한글만으로 잘 의미가 전달이 된다고 생각해서 괄호 없앴습니다.

```suggestion
- 이전의 코드 스니펫에서 사용되었던 `aria-pressed` 속성은 `"true"`라는 값을 가지는데, 이것은 `aria-pressed`가 `checked`가 있는 방식에서 진짜 불리언 속성이 아니기 때문입니다.
```

Reply:
반영 완료